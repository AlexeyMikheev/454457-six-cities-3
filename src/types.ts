import {OfferType} from "./consts";

export interface Location {
    latitude: number;
    longitude: number;
    zoom: number;
    center: number[];
}

export interface City {
    name: string;
    location: Location;
    center: number[];
    zoom: number;
}

export interface Owner {
    id: number;
    name: string;
    avatar: string;
    isTrust: boolean;
}

export interface Offer {
    owner: Owner;
    location: Location;
    city: City;
    id: number;
    isPremium: boolean;
    isMarked: boolean;
    cost: number;
    rating: number;
    name: string;
    description: string;
    roomsCount: number;
    membersCount: number;
    type: OfferType;
    image: string;
    images: string[];
    features: string[];
    cityName: string;
    center: number[];
    zoom: number;
}

export interface UserInfo {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface Review{
    id: number;
    comment: string;
    rating: number;
    date: number;
    user: Owner;
  }

export type HoveredChangeArgType = number | null;
export type SendCommentArgType = {
    rating: string;
    comment: string;
}
export type FormValueArgsType = {
    target: {
        name: string;
        value: string;
    };
}

export type SubmitArgsType = {
    login: string;
    password: string;
}

export interface FormState{
    [key: string]: string;
 }

export interface BooleanState{
    isToggled: boolean;
 }

export type OfferHoveredChangeFunction = (offerId: HoveredChangeArgType) => void | null;
export type OfferChangeFunction = (offerId: number) => void;
export type CityChangeFunction = (cityName: string) => void;
export type SendCommentFunction = (offerId: number, args: SendCommentArgType) => void;
export type FormValueChangeFunction = (args: FormValueArgsType) => void;
export type FormSubmitFunction = (args: SubmitArgsType) => void;

export type EmptyFunction = () => void;

export type MapSetting = {
    zoomControl: boolean;
    marker: boolean;
}
