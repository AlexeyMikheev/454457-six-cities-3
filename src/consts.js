
export const Feature = {
  WIFI: `Wi-Fi`,
  WASHINGMACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEEMACHINE: `Coffee machine`,
  BABYSET: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABELTV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const FEATURES = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

export const OfferType = {
  APARTMENT: `Apartment`,
  ROOM: `Private room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

export const OFFERTYPES = [
  OfferType.APARTMENT,
  OfferType.ROOM,
  OfferType.HOUSE,
  OfferType.HOTEL
];

export const ViewMode = {
  Main: 0,
  Property: 1
};

export const VIEWMODES = [
  ViewMode.Main,
  ViewMode.Property
];

export const MIN_RATING = 1;
export const MAX_RATING = 5;

export const MAX_REVIEWS_DISPLAY_COUNT = 10;

export const MAX_IMAGES_DISPLAY_COUNT = 6;
export const MAX_NEAR_DISPLAY_COUNT = 3;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LH: `Price: low to high`,
  PRICE_HL: `Price: high to low`,
  TOPRATED: `Top rated first`
};

export const SORTTYPES = [
  SortType.POPULAR,
  SortType.PRICE_LH,
  SortType.PRICE_HL,
  SortType.TOPRATED
];

export const RatingValues = [5, 4, 3, 2, 1];

export const ErrorType = {
  UNAUTHORIZED: 401,
  BABREQUEST: 400
};

export const TIMEOUT = 5000;
export const URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const AuthStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};

export const LoadingStatus = {
  DEFAULT: `DEFAULT`,
  SUCCESS: `SUCCESS`,
  LOADING: `LOADING`,
  ERROR: `ERROR`
};

export const LoadingStatuses = [
  LoadingStatus.DEFAULT,
  LoadingStatus.SUCCESS,
  LoadingStatus.LOADING,
  LoadingStatus.ERROR
];

export const Url = {
  COMMENTS: `comments`,
  HOTELS: `hotels`,
  LOGIN: `login`,
  NEARBY: `nearby`
};

export const AppRoute = {
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  ROOT: `/`,
};
