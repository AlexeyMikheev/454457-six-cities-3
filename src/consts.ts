export enum Feature {
    WIFI= `Wi-Fi`,
    WASHINGMACHINE= `Washing machine`,
    TOWELS= `Towels`,
    HEATING= `Heating`,
    COFFEEMACHINE= `Coffee machine`,
    BABYSET= `Baby seat`,
    KITCHEN= `Kitchen`,
    DISHWASHER= `Dishwasher`,
    CABELTV= `Cabel TV`,
    FRIDGE= `Fridge`
  }

export const FEATURES: Feature[] = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

export enum OfferType {
  APARTMENT= `Apartment`,
  ROOM= `Private room`,
  HOUSE= `House`,
  HOTEL= `Hotel`
}

export const OFFERTYPES: OfferType[] = [
  OfferType.APARTMENT,
  OfferType.ROOM,
  OfferType.HOUSE,
  OfferType.HOTEL
];

export enum ViewMode {
  Main= 0,
  Property= 1,
  Favorite= 2
}

// export const VIEWMODES: ViewMode[] = [
//   ViewMode.Main,
//   ViewMode.Property,
//   ViewMode.Favorite
// ];

export const MIN_RATING = 1;
export const MAX_RATING = 5;

export const MAX_REVIEWS_DISPLAY_COUNT = 10;

export const MAX_IMAGES_DISPLAY_COUNT = 6;
export const MAX_NEAR_DISPLAY_COUNT = 3;

export enum SortType {
  POPULAR= `Popular`,
  PRICE_LH= `Price: low to high`,
  PRICE_HL= `Price: high to low`,
  TOPRATED= `Top rated first`
}

export const SORTTYPES: SortType[] = [
  SortType.POPULAR,
  SortType.PRICE_LH,
  SortType.PRICE_HL,
  SortType.TOPRATED
];

export const RatingValues = [`5`, `4`, `3`, `2`, `1`];

export enum ErrorType {
  UNAUTHORIZED= 401,
  BABREQUEST= 400,
  NOTFOUND= 404
}

export const TIMEOUT = 5000;
export const URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export enum AuthStatus {
    NO_AUTH= `NO_AUTH`,
    LOADING= `LOADING`,
    AUTH= `AUTH`
  }

export enum LoadingStatus {
  DEFAULT= `DEFAULT`,
  SUCCESS= `SUCCESS`,
  LOADING= `LOADING`,
  ERROR= `ERROR`
}

export const LoadingStatuses: LoadingStatus[] = [
  LoadingStatus.DEFAULT,
  LoadingStatus.SUCCESS,
  LoadingStatus.LOADING,
  LoadingStatus.ERROR
];

export enum Url {
  COMMENTS= `comments`,
  HOTELS= `hotels`,
  FAVORITE= `favorite`,
  LOGIN= `login`,
  NEARBY= `nearby`
}

export enum AppRoute {
  LOGIN= `/login`,
  FAVORITES= `/favorites`,
  OFFER= `/offer`,
  ROOT= `/`,
}


export enum FavoriteState {
  MARKED= 1,
  UNMARKED= 0
}

export enum FavoriteButtonType {
  PROPERTY= 1,
  CARD= 0
}

// export const FavoriteButtonTypes = [
//   FavoriteButtonType.PROPERTY,
//   FavoriteButtonType.CARD
// ];

export enum ImageCardWidth {
  CARD_WIDTH= 260,
  CARD_HEIGHT= 200,
  FAVORITE_WIDTH= 150,
  FAVORITE_HEIGHT= 110
}

export const MIN_COMMENT_LENGTH = 50;
