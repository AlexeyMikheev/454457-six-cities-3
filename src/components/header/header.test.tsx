import * as React from "react";
import {mount} from "enzyme";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {AuthStatus} from "../../consts";
import {UserInfo} from "../../types";
import {BrowserRouter} from "react-router-dom";
import NameSpace from "../../reducer/name-space.js";
import Header from "./header";

const mockStore = configureStore([]);

const mockUserInfo: UserInfo = {
  [`avatar_url`]: `img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  [`is_pro`]: false,
  name: `Oliver.conner`,
};


it(`Header snapshot (AuthStatus.NO_AUTH)`, () => {

  const store = mockStore({[NameSpace.USER]: {
    authStatus: AuthStatus.NO_AUTH,
    authInfo: null
  }});

  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const headerComponent = mount(<BrowserRouter><Provider store={store}><Header /></Provider></BrowserRouter>, {attachTo: div});

  expect(headerComponent).toMatchSnapshot();
});

it(`Header snapshot (AuthStatus.AUTH)`, () => {

  const store = mockStore({[NameSpace.USER]: {
    authStatus: AuthStatus.AUTH,
    authInfo: mockUserInfo
  }});

  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const headerComponent = mount(<BrowserRouter><Provider store={store}><Header /></Provider></BrowserRouter>, {attachTo: div});

  expect(headerComponent).toMatchSnapshot();
});
