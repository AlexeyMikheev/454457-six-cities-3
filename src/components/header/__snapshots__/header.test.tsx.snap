// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot (AuthStatus.AUTH) 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          authInfo={
            Object {
              "avatar_url": "img/1.png",
              "email": "Oliver.conner@gmail.com",
              "id": 1,
              "is_pro": false,
              "name": "Oliver.conner",
            }
          }
          dispatch={[Function]}
          isAuthorized={true}
        >
          <header
            className="header"
          >
            <div
              className="container"
            >
              <div
                className="header__wrapper"
              >
                <div
                  className="header__left"
                >
                  <Link
                    className="header__logo-link"
                    to="/"
                  >
                    <LinkAnchor
                      className="header__logo-link"
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        className="header__logo-link"
                        href="/"
                        onClick={[Function]}
                      >
                        <img
                          alt="6 cities logo"
                          className="header__logo"
                          height="41"
                          src="/img/logo.svg"
                          width="81"
                        />
                      </a>
                    </LinkAnchor>
                  </Link>
                </div>
                <nav
                  className="header__nav"
                >
                  <ul
                    className="header__nav-list"
                  >
                    <li
                      className="header__nav-item user"
                    >
                      <Link
                        className="header__nav-link header__nav-link--profile"
                        to="/favorites"
                      >
                        <LinkAnchor
                          className="header__nav-link header__nav-link--profile"
                          href="/favorites"
                          navigate={[Function]}
                        >
                          <a
                            className="header__nav-link header__nav-link--profile"
                            href="/favorites"
                            onClick={[Function]}
                          >
                            <div
                              className="header__avatar-wrapper user__avatar-wrapper"
                            />
                            <span
                              className="header__user-name user__name"
                            >
                              Oliver.conner@gmail.com
                            </span>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
          </header>
        </Header>
      </Connect(Header)>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`Header snapshot (AuthStatus.NO_AUTH) 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          authInfo={null}
          dispatch={[Function]}
          isAuthorized={false}
        >
          <header
            className="header"
          >
            <div
              className="container"
            >
              <div
                className="header__wrapper"
              >
                <div
                  className="header__left"
                >
                  <Link
                    className="header__logo-link"
                    to="/"
                  >
                    <LinkAnchor
                      className="header__logo-link"
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        className="header__logo-link"
                        href="/"
                        onClick={[Function]}
                      >
                        <img
                          alt="6 cities logo"
                          className="header__logo"
                          height="41"
                          src="/img/logo.svg"
                          width="81"
                        />
                      </a>
                    </LinkAnchor>
                  </Link>
                </div>
                <nav
                  className="header__nav"
                >
                  <ul
                    className="header__nav-list"
                  >
                    <li
                      className="header__nav-item user"
                    >
                      <Link
                        className="header__nav-link header__nav-link--profile"
                        to="/login"
                      >
                        <LinkAnchor
                          className="header__nav-link header__nav-link--profile"
                          href="/login"
                          navigate={[Function]}
                        >
                          <a
                            className="header__nav-link header__nav-link--profile"
                            href="/login"
                            onClick={[Function]}
                          >
                            <div
                              className="header__avatar-wrapper user__avatar-wrapper"
                            />
                            <span
                              className="header__user-name user__name"
                            >
                              Sing in
                            </span>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
          </header>
        </Header>
      </Connect(Header)>
    </Provider>
  </Router>
</BrowserRouter>
`;
